### PR TITLE
Create home page

### PR DESCRIPTION
**Pull request by kravich2**

### CODE REVIEW

File: UWO.jpeg
Comment (varangill) [inline UWO.jpeg:1]: Are UWO.jpeg and UWO2.jpeg still used? If not, they can be removed, and if they are used, they should be in the `client` folder

File: client/package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "react": "^18.2.0",
     "react-bootstrap": "^2.9.1",
     "react-dom": "^18.2.0",
-    "react-router-dom": "^6.19.0"
+    "react-router-dom": "^6.21.1"
```
Comment (varangill) [inline client/package.json:18]: Any reason for a dependency update or was it an accident? not a big deal right now with how small our codebase is so I'm just curious

File: client/src/components/MyClubs.tsx [hunk 2]
```diff
@@ -1,10 +1,117 @@
...
 export default function MyClubs() {
+  const initialClubs = [
+    { name: 'Tennis Club', id: 'tennis' },
+    { name: 'Chess Club', id: 'chess' },
+    { name: 'Reading Club', id: 'reading' },
+    { name: 'Science Club', id: 'science' }
+  ];
+
+  const [clubs, setClubs] = useState(initialClubs);
+  const [leftClubs, setLeftClubs] = useState([]);
```
Comment (varangill) [inline client/src/components/MyClubs.tsx]: From my understanding, `leftClubs` is the list of clubs the user has recently left and we're showing an `undo` option for, right? Not every single club the user has left? If so, this variable name is a bit misleading, renaming it to something like `undoableClubs` would be better

File: client/src/components/MyClubs.tsx [hunk 3]
```diff
@@ -1,10 +1,117 @@
...
+    setLeftClubs([...leftClubs, leftClub]);
+
+    // Set a timer to automatically remove the undo option after 30 seconds
+    setTimeout(() => {
+      setLeftClubs(leftClubs => leftClubs.filter(club => club.id !== clubId));
+    }, 5000); // 30 seconds
+  };
+
+  // Function to handle undoing leave action
+  const undoLeave = (clubId) => {
```
Comment (varangill) [inline client/src/components/MyClubs.tsx]: Do we really want an undo option for this release? We can discuss it as a group but I think it's something to work on if we have extra time at the end of the year, there's more key functionalities to focus on currently IMO, just leaving a club is fine for now

File: client/src/components/MyClubs.tsx [hunk 4]
```diff
@@ -1,10 +1,117 @@
...
+  };
+
+  // Filter clubs based on query
+  const filteredClubs = clubs.filter(club =>
+    club.name.toLowerCase().includes(query.toLowerCase())
+  );
+
   return (
-    <div>
+    <div style={{ textAlign: 'center' }}>
```
Comment (varangill) [inline client/src/components/MyClubs.tsx]: Can you move all in-line style instances like this to a CSS file? @tmohaupt has [an example of moving styles to a CSS file here](https://github.com/varangill/ClubHub/pull/9/commits/7d70e4a6c0551b49dd66e341acd208e4e6e3c6fc) if you need help

There's also in-line styles on lines 45, 49, 61, 65, and the three style constants at the bottom of the file that could be removed to clean up the file a bit

File: package.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
```
Comment (varangill) [inline package.json]: I assume this was an accident but we don't want `package.json` or `package-lock.json` in the uppermost folder, these can be removed

**Review by varangill — CHANGES_REQUESTED** [review]
Looks pretty good so far, few suggestions/comments though

File: client/src/components/MyClubs.tsx (hunk 3 above)
Comment (tmohaupt) [reply client/src/components/MyClubs.tsx]: I don't see any need for this feature. You can simply just rejoin from the AllClubs page

Comment (kravich2) [reply client/src/components/MyClubs.tsx]: I thought that if the user accidentally clicked the leave button for the wrong club than it would be easier to have an undo feature instead of trying to search for the club in the AllClubs pages.

File: UWO.jpeg
Comment (kravich2) [reply UWO.jpeg:1]: I removed the photos and have it refer to a URL now.

File: package.json (hunk 5 above)
Comment (kravich2) [reply package.json]: Removed

File: client/package.json (hunk 1 above)
Comment (kravich2) [reply client/package.json:18]: it was an accident.

File: client/src/components/MyClubs.tsx (hunk 2 above)
Comment (kravich2) [reply client/src/components/MyClubs.tsx]: Renamed.

File: client/src/components/MyClubs.tsx (hunk 3 above)
Comment (varangill) [reply client/src/components/MyClubs.tsx]: I'd agree with Tyler, don't think we need this tbh

File: server/env [hunk 6]
```diff
@@ -0,0 +1,5 @@
+USER=postgres
```
Comment (varangill) [inline server/env]: This file should be called `.env` (with the dot) and should not be included in the commit, your issues running the server were probably caused by you naming the file `env`

File: client/src/components/MyClubs.tsx (hunk 4 above)
Comment (varangill) [reply client/src/components/MyClubs.tsx]: Still want this done for this PR @kravich2

File: client/src/components/MyClubs.tsx (hunk 3 above)
Comment (varangill) [reply client/src/components/MyClubs.tsx]: Gonna say we can just remove the undo stuff from the PR

File: server/package.json [hunk 7]
```diff
@@ -22,6 +22,7 @@
   "dependencies": {
     "dotenv": "^16.3.1",
     "express": "^4.17.1",
+    "knex": "^3.1.0",
```
Comment (varangill) [inline server/package.json]: You should try to install knex globally instead of adding a dependency for only your machine, if you're having troubles and you can't find a solution online then ask in the group chat for help

File: server/src/database/knexfile.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (varangill) [inline server/src/database/knexfile.js]: We have a knexfile already in Typescript

File: server/src/database/migrations/20240108011223_migration_name.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+/**
```
Comment (varangill) [inline server/src/database/migrations/20240108011223_migration_name.js]: Remove this empty migration

File: client/env [hunk 10]
```diff
@@ -0,0 +1 @@
+VITE_BACKEND_URL=localhost:3000
```
Comment (varangill) [inline client/env]: Same as other env, should be `.env` and not in the commit

**Review by varangill — CHANGES_REQUESTED** [review]
More comments

Also it's good practice to commit frequently and base commits on smaller, discrete changes instead of one bigger commit every time you want to push up your changes, for example one small commit from this giant commit could've been "Delete unnecessary files" which deleted the two .jpeg files and package.json and package-lock.json

File: server/src/database/migrations/20240108011223_migration_name.js (hunk 9 above)
Comment (kravich2) [reply server/src/database/migrations/20240108011223_migration_name.js]: Removed.

File: server/src/database/knexfile.js (hunk 8 above)
Comment (kravich2) [reply server/src/database/knexfile.js]: Removed.

File: client/env (hunk 10 above)
Comment (kravich2) [reply client/env]: Thanks for telling me, I'll try it out now.

File: server/env (hunk 6 above)
Comment (kravich2) [reply server/env]: Thanks for telling me, I'll try it now

File: client/src/components/MyClubs.tsx (hunk 3 above)
Comment (kravich2) [reply client/src/components/MyClubs.tsx]: I removed it

File: client/src/components/MyClubs.tsx (hunk 4 above)
Comment (kravich2) [reply client/src/components/MyClubs.tsx]: I entered the changes.

File: .gitignore [hunk 11]
```diff
@@ -22,4 +22,6 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
-.env
\ No newline at end of file
+.env
+server/env
+client/env
```
Comment (varangill) [inline .gitignore]: like I mentioned in the previous review, you shouldn't have any files called `env`, only `.env`, this doesn't need to be in the .gitignore

File: client/src/components/MyClubs.tsx [hunk 12]
```diff
@@ -1,10 +1,62 @@
-import NavBar from "./NavigationBar";
+import React, { useState } from 'react';
+import NavBar from './NavigationBar';
+import { Link } from 'react-router-dom';
+import '../MyClubs.css'; // Import the CSS file
```
Comment (varangill) [inline client/src/components/MyClubs.tsx]: import this in App.tsx like the other CSS file is imported

File: server/package.json (hunk 7 above)
Comment (varangill) [reply server/package.json]: Still want this uninstalled from here and package-lock.json

File: client/src/components/AllClubs.tsx [hunk 13]
```diff
@@ -1,10 +1,152 @@
...
   return (
-    <div>
+    <div style={{ textAlign: 'center' }}>
       <NavBar />
-      All Clubs
+
+      {/* Total Club Count and Search Bar */}
+      <div
+        style={{
+          position: 'absolute',
```
Comment (varangill) [inline client/src/components/AllClubs.tsx]: Lots of styling in this file that should be moved to a new CSS file like the other

**Review by varangill — CHANGES_REQUESTED** [review]
Still more changes to make - read my review comments carefully please, same thing with unclear commit messages as last time

File: server/package.json (hunk 7 above)
Comment (kravich2) [reply server/package.json]: Uninstalled it

File: .gitignore (hunk 11 above)
Comment (kravich2) [reply .gitignore]: Deleted the .env files

File: client/src/components/AllClubs.tsx (hunk 13 above)
Comment (kravich2) [reply client/src/components/AllClubs.tsx]: Added

File: client/src/components/MyClubs.tsx (hunk 12 above)
Comment (kravich2) [reply client/src/components/MyClubs.tsx]: added

File: client/src/components/ClubDetails.tsx [hunk 14]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (varangill) [inline client/src/components/ClubDetails.tsx]: What's the purpose of this component? It isn't used anywhere? It looks like it'd serve a similar purpose as `ClubPage` right?

File: client/src/components/ClubDetails.tsx [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+  const { clubId } = useParams();
+
+  // You can use the clubId to fetch the club details or display them as needed
+  // Example:
+  // const clubDetails = fetchClubDetails(clubId);
+
+  return (
+    <div>
+      <h2>Club Details Page</h2>
+      <p>Club ID: {clubId}</p>
```
Comment (varangill) [inline client/src/components/ClubDetails.tsx]: Data object IDs should be for internal use only, they don't need to know the club ID on the UI page

File: client/src/components/ClubPage.tsx [hunk 16]
```diff
@@ -1,10 +1,45 @@
-import NavBar from "./NavigationBar";
+import React from 'react';
+import NavBar from './NavigationBar';
+import { Link, useParams } from 'react-router-dom';
 
 export default function ClubPage() {
+  // Get the clubName parameter from the URL
+  const { clubName } = useParams();
```
Comment (varangill) [inline client/src/components/ClubPage.tsx]: The club name isn't in parameters, if you want the club name you need to take the ID (which would be in the params) and query the API endpoint for club info. But for now I'd just remove this and use some placeholder club name, this PR is already huge, you can add the API queries in a future PR, it's better to just have smaller PRs

File: client/src/components/ClubPage.tsx [hunk 17]
```diff
@@ -1,10 +1,45 @@
...
+      {/* Display Club Name */}
+      <h2
+        style={{
+          fontSize: '36px',
+          color: '#333',
+          margin: '20px 0',
+          fontFamily: 'Arial, sans-serif'
+        }}
+      >
+        Club page - {clubName ? decodeURIComponent(clubName) : 'Unknown Club'}
```
Comment (varangill) [inline client/src/components/ClubPage.tsx]: Weird ChatGPT'd code... Not something we'd need

File: client/src/components/ClubPage.tsx [hunk 18]
```diff
@@ -1,10 +1,45 @@
...
+  const { clubName } = useParams();
+
   return (
-    <div>
+    <div style={{ textAlign: 'center' }}>
       <NavBar />
-      Club page
+      {/* Display Club Name */}
+      <h2
+        style={{
```
Comment (varangill) [inline client/src/components/ClubPage.tsx]: Styling goes in CSS file, not in-line

File: client/src/components/ClubPage.tsx [hunk 19]
```diff
@@ -1,10 +1,45 @@
...
+      {/* Add a Go Back button */}
+      <Link to="/my_clubs" style={buttonStyle}>
+        Go Back
+      </Link>
     </div>
   );
 }
+
+// Button style for Go Back button
+const buttonStyle = {
```
Comment (varangill) [inline client/src/components/ClubPage.tsx]: More styling that should go in a CSS file

File: client/src/components/MyClubs.tsx (hunk 12 above)
Comment (varangill) [inline client/src/components/MyClubs.tsx]: You're still importing this CSS file here

File: client/src/components/MyClubs.tsx [hunk 20]
```diff
@@ -1,10 +1,62 @@
...
+
+      {/* Display Active Clubs */}
+      <table className="clubs-table"> {/* Use CSS class for styling */}
+        <tbody>
+          {filteredClubs.map((club) => (
+            <tr key={club.id}>
+              <td>
+                {/* Use Link to navigate to ClubPage with the club's ID and name */}
+                <Link
+                  to={`/club/${club.id}?clubName=${encodeURIComponent(club.name)}`}
```
Comment (varangill) [inline client/src/components/MyClubs.tsx]: Club name shouldn't be in the URL like this, more weird ChatGPT'd code

File: client/src/components/MyClubs.tsx [hunk 21]
```diff
@@ -1,10 +1,62 @@
...
-      My Clubs
+
+      <h2 className="club-heading">My Clubs</h2> {/* Use CSS class for styling */}
+
+      <input
+        type="text"
+        placeholder="Search clubs..."
+        value={query}
+        onChange={(e) => setQuery(e.target.value)}
+        className="search-input" // Use CSS class for styling
```
Comment (varangill) [inline client/src/components/MyClubs.tsx]: These "Use CSS class for styling" comments are a bit odd, they're not really blockers but they don't serve any purpose? It's known that the `className` is used as a CSS selector, that's how classes and CSS work

File: client/src/components/AllClubs.tsx [hunk 22]
```diff
@@ -1,10 +1,97 @@
-import NavBar from "./NavigationBar";
+import React, { useState } from 'react';
+import NavBar from './NavigationBar';
+import { Link } from 'react-router-dom';
+import '../AllClubs.css'; // Import the CSS file
```
Comment (varangill) [inline client/src/components/AllClubs.tsx]: Import this in through App.tsx like other CSS files

File: client/src/components/AllClubs.tsx [hunk 23]
```diff
@@ -1,10 +1,97 @@
...
+        className="search-input"
+      />
+
+      <table className="clubs-table">
+        <tbody>
+          {filteredClubs.map((club) => (
+            <tr key={club.id}>
+              <td>
+                <Link
+                  to={`/club/${club.id}?clubName=${encodeURIComponent(
```
Comment (varangill) [inline client/src/components/AllClubs.tsx]: Same as my comment in MyClubs.tsx - club name isn't supposed to be like this

File: client/src/components/AllClubs.tsx [hunk 24]
```diff
@@ -1,10 +1,97 @@
...
+        type="text"
+        placeholder="Search clubs..."
+        value={query}
+        onChange={(e) => setQuery(e.target.value)}
+        className="search-input"
+      />
+
+      <table className="clubs-table">
+        <tbody>
+          {filteredClubs.map((club) => (
```
Comment (varangill) [inline client/src/components/AllClubs.tsx]: This club listing is remarkably similar to the one in `MyClubs.tsx` and you could just use a common component between the two to get rid of repetitive code, but this is fine for now, it can be changed in a future PR

File: .gitignore (hunk 11 above)
Comment (varangill) [inline .gitignore]: Delete these `env` files from the .gitignore too please

File: client/src/components/AllClubs.tsx [hunk 25]
```diff
@@ -1,10 +1,97 @@
...
+                >
+                  {club.name}
+                </Link>
+                <button className="join-button" onClick={() => joinClub(club.id)}>
+                  Join
+                </button>
+              </td>
+            </tr>
+          ))}
+          {joinedClubs.map((club) => (
```
Comment (varangill) [inline client/src/components/AllClubs.tsx]: With the way this is formatted, you're going to have every possible club listed out first (including ones you're in), then you'll have every club you're in, so you'll have duplicate clubs and users will see a "Join" button on clubs they're already in.

IMO you can just remove join buttons from this AllClubs page and move it to the actual specific club page, so users have to click onto a club's page before joining it. Then you can get rid of `joinedClubs`. It'll be a lot cleaner and an easier fix to the above issue

**Review by varangill — CHANGES_REQUESTED** [review]
A lot of the stuff in this review, particularly the CSS stuff, is stuff I mentioned in a previous review on this exact same PR. Please try to keep previous feedback in mind when coding

File: client/src/components/ClubDetails.tsx (hunk 15 above)
Comment (kravich2) [reply client/src/components/ClubDetails.tsx]: I had the data object id for when I was trying to show the detail description of the club when the user would click the button. I'm still working on the function which is why it is still there.

File: client/src/components/ClubDetails.tsx (hunk 14 above)
Comment (kravich2) [reply client/src/components/ClubDetails.tsx]: Took it out.

File: client/src/components/ClubPage.tsx (hunk 17 above)
Comment (kravich2) [reply client/src/components/ClubPage.tsx]: taken out

File: client/src/components/ClubPage.tsx (hunk 18 above)
Comment (kravich2) [reply client/src/components/ClubPage.tsx]: Done.

File: client/src/components/ClubPage.tsx (hunk 16 above)
Comment (kravich2) [reply client/src/components/ClubPage.tsx]: Deleted.

**Review by varangill — APPROVED** [review]
